core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -34,12 +36,53 @@ abstract class SparkFunSuite
   with Logging {
 // scalastyle:on
 
+  val threadWhiteList = Set(
+    /**
+     * Netty related threads.
+     */
+    "netty.*",
+
+    /**
+     * A Single-thread singleton EventExecutor inside netty which creates such threads.
+     */
+    "globalEventExecutor.*",
+
+    /**
+     * Netty creates such threads.
+     * Checks if a thread is alive periodically and runs a task when a thread dies.
+     */
+    "threadDeathWatcher.*",
+
+    /**
+     * These threads are created by spark when internal RPC environment initialized and later used.
```

Comment:
Yes, but why is it ok to whitelist them?

Reply:
These threads started when SparkContext created and remain there. These can be destroyed when sc.stop() called but I've seen this pattern rarely. If we would like to follow this, huge amount of tests should be modified and I don't see the ROI. On the other hand not whitelisting them would trigger false positives when context tried to be reused like you mentioned in case of hive. Ideas/opinions? Shall we drop them?

Reply:
> These can be destroyed when sc.stop() called but I've seen this pattern rarely.

Not sure I understand. Tests should be stopping their contexts (sql and hive tests notwithstanding), and if these threads are not going away when that happens, it's a bug.

After all, that's the kind of thing I expect this code to be catching.

Reply:
Fully agree with your stand and tests should stop their context. What I see is that majority of tests are just not doing that. OK, then I suggest to remove them from the whitelist. Agreed?

Reply:
Sounds like the case (removing from whitelist), but if tests really weren't stopping their contexts, I'd expect other errors to happen. Do you have examples of tests that show this issue?

Reply:
TaskManagerSuite.test("TaskSet with no preferences") prints this out:

===== POSSIBLE THREAD LEAK IN SUITE o.a.s.scheduler.TaskSetManagerSuite, thread names: rpc-client-1-1, rpc-server-3-2, rpc-client-1-4, rpc-server-3-6, rpc-client-1-3, rpc-server-3-7, rpc-server-3-1, rpc-server-3-8, rpc-server-3-4, rpc-client-1-7, rpc-client-1-6, rpc-client-1-2, rpc-server-3-5, shuffle-client-4-1, shuffle-server-5-1, rpc-server-3-3, rpc-client-1-5, rpc-client-1-8 =====

This is the result even if the test contains only this line:

sc = new SparkContext("local", "test")

Reply:
`TaskSetManagerSuite` extends `LocalSparkContext` which stops contexts in its `afterEach` method, so if the context is not being stopped, or the threads are not being stopped, there's an issue somewhere.

Reply:
Ahhh, that's a key. I've no doubt these whitelist items should be removed. Thanks!

Reply:
Removed.

Reply:
In the meantime analyzed the related threads and here are the findings:

    /**
     * During [[SparkContext]] creation [[org.apache.spark.storage.BlockManager]]
     * creates event loops. One is wrapped inside
     * [[org.apache.spark.network.server.TransportServer]]
     * the other one is inside [[org.apache.spark.network.client.TransportClient]].
     * The thread pools behind shut down asynchronously triggered by [[SparkContext#close]].
     * Manually checked and all of them stopped properly.
     */
    "shuffle-client.*",
    "shuffle-server.*"

Do you think in this situation they can be added to the whitelist?